sdk/core/System.ClientModel/src/Convenience/ClientConnection.cs

```diff
@@ -0,0 +1,130 @@
+    public ClientConnection(string id, string locator, object credential)
+    {
+        if (string.IsNullOrWhiteSpace(id))
+            throw new ArgumentException("Id cannot be null or empty.", nameof(id));
+        if (string.IsNullOrWhiteSpace(locator))
+            throw new ArgumentException("Locator cannot be null or empty.", nameof(locator));
+        if (credential is null)
+            throw new ArgumentNullException(nameof(credential), "Credential cannot be null for Entra ID authentication.");
+
+        Id = id;
+        Locator = locator;
+        Authentication = ClientAuthenticationMethod.EntraId;
+        Credential = credential;
+        ApiKeyCredential = null;
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="ClientConnection"/> struct with a with no authentication.
+    /// </summary>
+    /// <param name="id">The identifier for the connection.</param>
+    /// <param name="locator">The endpoint or resource identifier.</param>
+    public ClientConnection(string id, string locator)
+    {
+        if (string.IsNullOrWhiteSpace(id))
+            throw new ArgumentException("Id cannot be null or empty.", nameof(id));
+        if (string.IsNullOrWhiteSpace(locator))
+            throw new ArgumentException("Locator cannot be null or empty.", nameof(locator));
+
+        Id = id;
+        Locator = locator;
+        Authentication = ClientAuthenticationMethod.NoAuth;
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="ClientConnection"/> struct with the specified subclient ID.
+    /// It is only for the JSON serializer.
+    /// </summary>
+    /// <param name="id"></param>
+    /// <param name="locator">The subclient ID.</param>
+    /// <param name="auth"></param>
+    internal ClientConnection(string id, string locator, ClientAuthenticationMethod auth)
+    {
+        Id = id;
+        Locator = locator;
+        Authentication = auth;
+    }
+
+    /// <summary>
+    /// Gets the kind of connection used by the client.
+    /// </summary>
+    public ClientAuthenticationMethod Authentication { get; }
+
+    /// <summary>
+    /// Gets the connection identifier.
+    /// </summary>
+    public string Id { get; }
+
+    /// <summary>
+    /// This is either URI, name, or something similar.
+    /// </summary>
+    public string Locator { get; }
+
+    /// <summary>
+    /// Gets the API key credential, if applicable.
+    /// </summary>
+    public string? ApiKeyCredential { get; }
+
+    /// <summary>
+    /// Gets the token credential, if applicable.
+    /// Since TokenCredential is not available in this package, using object type as a placeholder.
+    /// </summary>
+    public object? Credential { get; }
+
+    /// <summary>
+    /// Converts the connection to a URI.
+    /// </summary>
+    /// <returns>A URI representation of the connection.</returns>
+    public Uri LocatorAsUri()
```

Comment:
should this be `public bool TryGetLocatorAsUri(out Uri)` ?

Reply:
I am fine with that as future proofing change. As these APIs are used today, this should never fail or something is very wrong with the data, e.g. if the method that calls this is an extension method creating a client that needs a URI, it better casts to URI. If the method is creating a client that has a name (string), e.g. serviceBusClient.GetSender, the method better know not call LocatorAsUri because SB sender locator is not a URI.